packages/backend-for-frontend/test/validateTokenGeneration.test.ts

```diff
@@ -0,0 +1,515 @@
+            kind: authorizationApi.ClientKind.enum.API,
+          },
+        }),
+      },
+      client: {
+        getClientKeyById: vi.fn().mockResolvedValue({}),
+      },
+    },
+  } as unknown as PagoPAInteropBeClients;
+
+  beforeEach(() => {
+    setupValidationMocks();
+  });
+
+  function setupValidationMocks(): void {
+    vi.spyOn(
+      clientAssertionValidation,
+      "validateRequestParameters"
+    ).mockReturnValue({
+      errors: undefined,
+      data: {
+        client_assertion: MOCK_CLIENT_ASSERTION,
+        client_assertion_type: MOCK_CLIENT_ASSERTION_TYPE,
+        grant_type: MOCK_GRANT_TYPE,
+        client_id: MOCK_CLIENT_ID,
+      },
+    });
+
+    vi.spyOn(
+      clientAssertionValidation,
+      "verifyClientAssertion"
+    ).mockReturnValue({
+      errors: undefined,
+      data: {
+        header: { kid: MOCK_KID, alg: "RS256", typ: "JWT" },
+        payload: MOCK_JWT_PAYLOAD,
+      },
+    });
+
+    vi.spyOn(
+      clientAssertionValidation,
+      "verifyClientAssertionSignature"
+    ).mockResolvedValue({
+      errors: undefined,
+      data: {
+        ...MOCK_JWT_PAYLOAD,
+        purposeId: "purpose-id-123",
+      },
+    });
+  }
+
+  const toolService = toolsServiceBuilder(mockClients);
+
+  describe("Success case", () => {
+    it("should validate generated token successfully", async () => {
+      const validationResult = await toolService.validateTokenGeneration(
+        MOCK_CLIENT_ID,
+        MOCK_CLIENT_ASSERTION,
+        MOCK_CLIENT_ASSERTION_TYPE,
+        MOCK_GRANT_TYPE,
+        bffMockContext
+      );
+
+      expect(validationResult).toMatchObject({
+        eservice: undefined,
+        steps: {
+          clientAssertionValidation: {
+            result: bffApi.TokenGenerationValidationStepResult.Enum.PASSED,
+            failures: [],
+          },
+          publicKeyRetrieve: {
+            result: bffApi.TokenGenerationValidationStepResult.Enum.PASSED,
+            failures: [],
+          },
+          clientAssertionSignatureVerification: {
+            result: bffApi.TokenGenerationValidationStepResult.Enum.PASSED,
+            failures: [],
+          },
+          platformStatesVerification: {
+            result: bffApi.TokenGenerationValidationStepResult.Enum.PASSED,
+            failures: [],
+          },
+        },
+      });
+    });
+  });
+
+  describe("Failure cases", () => {
+    it("should handle parameters validation errors", async () => {
+      const parameterValidationError = {
+        code: "invalidAssertionType" as const,
+        title: "Invalid assertion type",
+        detail: `Assertion type ${MOCK_CLIENT_ASSERTION_TYPE} is invalid. Expected: urn:ietf:params:oauth:client-assertion-type:jwt-bearer`,
+        name: "Error",
+        message: `Assertion type ${MOCK_CLIENT_ASSERTION_TYPE} is invalid. Expected: urn:ietf:params:oauth:client-assertion-type:jwt-bearer`,
+        errors: [],
+      };
```

Comment:
```suggestion
      const parameterValidationError: ApiError<"invalidAssertionType"> = {
        code: "invalidAssertionType",
        title: "Invalid assertion type",
        detail: `Assertion type ${MOCK_CLIENT_ASSERTION_TYPE} is invalid. Expected: urn:ietf:params:oauth:client-assertion-type:jwt-bearer`,
        name: "Error",
        message: `Assertion type ${MOCK_CLIENT_ASSERTION_TYPE} is invalid. Expected: urn:ietf:params:oauth:client-assertion-type:jwt-bearer`,
        errors: [],
      };
```

To avoid using `as const` and to have it completely typed